### ISSUE TITLE
[Bug]: Auto focus not working anymore in modal view

### ISSUE THREAD

**Issue by aivanis** [appsmithorg/appsmith#19957]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have v1.8.14.1 running on my server. When I add a input field with Auto Focus within a modal view the field is focused when I open the modal.
I tried updating to v1.9.3.2 and this doesn't work any more. Same for the cloud hosting in the sample app I created it doesn't work any more.

### Steps To Reproduce

1. Click on Submit, the modal will open but the field is not focused

### Public Sample App

https://app.appsmith.com/app/untitled-application-1/page1-635160d1e15bd110fc705f51

### Version

both cloud and self-hosted v1.9.3.2

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_715e4)

**Comment by somangshu** [comment]
@aivanis thanks for reporting this issue 

This works as usual on the canvas. No changes observed in the input widget as well. This is happening specifically on a modal. cc @rahulramesha possible to help here? Please let me know otherwise

**Comment by honoru-washswat** [comment]
https://discord.com/channels/725602949748752515/1068125584602959963

I'm in the same situation please help!!

**Comment by dilippitchika** [comment]
@honoru-washswat we are already working on it

**Comment by rahulramesha** [comment]
@somangshu was looking into this. This issue seems to have Originated from the keyboard Accessibility PR, https://github.com/appsmithorg/appsmith/pull/18991
cc: @jsartisan

**Comment by vsvamsi1** [comment]
Hi @jsartisan, it is originating from this focus statement https://github.com/appsmithorg/appsmith/blob/release/app/client/src/widgets/ModalWidget/component/index.tsx#L175. If we remove that line the autofocus works as normal, Can we remove that line? was there any usecase you wanted to cover from that statement?
cc @somangshu

**Comment by dilippitchika** [comment]
@vsvamsi1 this might have been done to ensure the focus doesn't move out of modal ie focus trapping. Can you remove the code and check?

**Comment by vsvamsi1** [comment]
Hi @dilippitchika, yes it is working without it and as you have mentioned @jsartisan wanted to cover that usecase. So we have a fix https://github.com/appsmithorg/appsmith/pull/20244 to support it but we wanted to get it reviewed by Abhinav once.
